man/adoc/bpftrace.adoc

```diff
@@ -0,0 +1,2213 @@
+Values in the range `(-inf, min)` and `(max, inf)` get their get their own bucket too, bringing the total amount of buckets created to `M+2`.
+
+----
+i:ms:1 {
+  @ = lhist(rand %10, 0, 10, 1);
+}
+
+i:s:5 {
+  exit();
+}
+----
+
+Prints:
+
+----
+@:
+[0, 1)               306 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@         |
+[1, 2)               284 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@            |
+[2, 3)               294 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@          |
+[3, 4)               318 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@       |
+[4, 5)               311 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@        |
+[5, 6)               362 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@|
+[6, 7)               336 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@    |
+[7, 8)               326 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@      |
+[8, 9)               328 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@     |
+[9, 10)              318 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@       |
+----
+
+[#functions-max]
+=== max(int64 n)
+
+Update the map with `n` if `n` is bigger than the current value held.
+
+[#functions-min]
+=== min(int64 n)
+
+Update the map with `n` if `n` is smaller than the current value held.
+
+[#functions-stats]
+=== stats(int64 n)
+
+`stats` combines the `count`, `avg` and `sum` calls into one.
+
+----
+kprobe:vfs_read {
+  @bytes[comm] = stats(arg2);
+}
+----
+
+----
+@bytes[bash]: count 7, average 1, total 7
+@bytes[sleep]: count 5, average 832, total 4160
+@bytes[ls]: count 7, average 886, total 6208
+@
+----
+
+[#functions-sum]
+=== sum(int64 n)
+
+Calculate the sum of all `n` passed.
+
+[#functions-zero]
+=== zero(map m)
+
+*async*
+
+Set all values for all keys to zero.
+
+== Functions
+
+Functions that are marked *async* are asynchronous which can lead to unexpected behaviour, see the <<sync and async>> section for more information.
+
+*compile time* functions are evaluated at compile time, a static value will be compiled into the program.
+
+*unsafe* functions can have dangerous side effects and should be used with care, the `--unsafe` flag is required for use.
+
+[#functions-buf]
+=== buf_t buf(void * data, [int64 length])
+
+`buf` reads `length` amount of bytes from address `data`.
+The maximum value of `length` is limited to the `BPFTRACE_STRLEN` variable.
+For arrays the `length` is optional, it is automatically inferred from the signature.
+
+`buf` is address space aware and will call the correct helper based on the address space associated with `data`.
+
+The `buf_t` object returned by `buf` can safely be printed as a hex encoded string with the `%r` format specifier.
+
+----
+i:s:1 {
+  printf("%r\n", buf(kaddr("avenrun"), 8));
+}
+----
+
+----
+\x00\x03\x00\x00\x00\x00\x00\x00
+\xc2\x02\x00\x00\x00\x00\x00\x00
+----
+
+[#functions-cat]
+=== void cat(string namefmt, [...args])
```

Comment:
When compiled to html, I find the full function signature in the heading a ugly to read in the table of contents. Any thoughts on leaving the heading `cat()` and providing full signature(s) in the body?

Reply:
yeah, applied